app/models/taxon_search_results.rb

```diff
@@ -8,9 +8,12 @@ def initialize(search_response)
   def taxons
     @taxons ||=
       begin
-        search_response['results'].map do |taxon_hash|
+        results = search_response['results'].map do |taxon_hash|
+          next if taxon_hash['publication_state'] == 'unpublished'
```

Comment:
What do you think of writing a test case on `taxon_search_results_spec.rb` to make sure this line doesn't get deleted in the future?